src/components/review/ReviewConfiguration.jsx

```diff
@@ -58,8 +62,86 @@ const ReviewDescriptionList = ({ children }) => {
     );
 };
 
-export const ReviewConfiguration = ({ deviceData, diskSelection, language, localizationData, requests, idPrefix, setIsFormValid, storageScenarioId, accounts }) => {
+const ChangeHostname = ({ initHostname }) => {
+    const [currentHostname, setCurrentHostname] = useState("");
+    const [error, setError] = useState([]);
+    const [isModalOpen, setIsModalOpen] = React.useState(false);
+
+    const validateHostname = (value) => {
+        const error = [];
+        if (value.length > 64) { error.push(_("Real host name must be 64 characters or less")) }
+        if (value.match(/[.a-z0-9-]*/)[0] !== value || value.indexOf("..") !== -1) { error.push(_("Real host name can only contain lower-case characters, digits, dashes, and periods (with populated subdomains)")) }
+
+        setError(error);
+    };
+
+    const onHostnameChanged = (value) => {
+        validateHostname(value);
+        setCurrentHostname(value);
+    };
+    const onSubmit = (event) => {
+        setHostname({ hostname: currentHostname })
+                .then(() => {
+                    handleModalToggle();
+                })
+                .catch(ex => {
+                    console.info({ ex });
+                });
+
+        if (event) { event.preventDefault() }
+        return false;
+    };
+
+    const handleModalToggle = _event => {
+        setIsModalOpen(!isModalOpen);
+    };
+
+    const disabled = error.length || (initHostname === currentHostname);
+    return (
+        <>
+            <Button
+              id="system_information_hostname_button" variant="link"
+              onClick={handleModalToggle}
+              isInline aria-label="edit hostname">
+                {_("edit")}
+            </Button>
+            <Modal
+              footer={
+                  <>
+                      <Button variant="primary" isDisabled={disabled} onClick={onSubmit}>{_("Change")}</Button>
+                      <Button variant="link" onClick={handleModalToggle}>{_("Cancel")}</Button>
+                  </>
+              }
+              id="system_information_change_hostname"
+              isOpen={isModalOpen}
+              onClose={handleModalToggle}
+              position="top"
+              title={_("Change host name")}
```

Comment:
Please ask in the channel about anaconda's host name vs hostname convention.

Reply:
We had a discussion about this in Slack a while back; I don't remember what the conclusion was... except that we should be consistent. But, then, consistent with what?

So, to capture what I remember of the discussion here....

Let's look at what we ship:

- GNOME uses "Device name".
- KDE... ????? (I can't find it anywhere, oddly. I've looked all over the place.)
- Linux system uses `/etc/hostname` and the `hostname` command. These are from the 1970s.
- Cockpit uses "Hostname" (one word), but this is a reflection of the above UNIX references.

Let's look at what everyone else is doing:

- Windows uses "Device name" (Windows 11); they previously used "Computer name" (Windows 10 and prior)
- macOS uses "Computer name"
- Android uses "Device name"
- iPhone uses "Name" (under General > About), but the phone is referred to "Device" in several other places, and when the name is being referred to in the generic sense, "Device name" is used.

As the most recent versions of GNOME, Windows, Android, and iOS all use "Device name" and also because it's a non-jargon name, I'd ideally prefer if we used "Device name". "Computer" name is similar, but more specific... as Fedora is used on tablets and [gaming consoles](https://bazzite.gg/) as well (which are definitely devices, but not considered "computers"). So, "Device" is more common and inclusive and the better of the two.

However, for historical reasons, "Hostname" is used by UNIX and Linux (which is why it's been referred to as "Hostname" in Cockpit too). It's not "Host name" (two words); it's "Hostname" (one word). But this is computer jargon, and interfaces should generally avoid jargon, except where it's extremely commonplace or unavoidable.

So the question then would be: Should we be more consistent with modern computing in general — using "Device name" — or be consistent with jargon from the 1970s :wink: and go with "Hostname"?

So, saying all that, **in summary**; we have two options: "Device name" (ideally) or "Hostname", _but not "Host name"._

We also talked about where this could be changed. Everything other than the review page would require a new screen. The review screen could be inline editing (which is in PatternFly, but usually not in Anaconda or Cockpit) or a modal (as it does in this PR). We did talk about it possibly also being in a networking page (when we have one, and where we'd have on — like on a server install) or even something like a username page.

https://redhat-internal.slack.com/archives/C04TE3FPD32/p1709802956865689?thread_ts=1709717365.590469&cid=C04TE3FPD32